client.go

```diff
@@ -33,7 +33,10 @@ func NewClient(ifi *net.Interface) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	return NewClientPacketConn(ifi, p)
+}
 
+func NewClientPacketConn(ifi *net.Interface, p net.PacketConn) (*Client, error) {
```

Comment:
I see. My feedback was about a generic Go style issue I noticed. I think only the project owner will be able to tell whether this change makes sense/belongs here at a higher level.